File: connectors/src/main/java/com/binance4j/spot/param/AllOrdersParams.java
package com.binance4j.spot.param;

import com.binance4j.core.annotation.Param;
import com.binance4j.core.param.Params;
import com.binance4j.spot.client.SpotClient;

/**
 * [SpotClient.getAllOrders] params.
 * 
 * @param symbol  Trading pair we want the orders.
 * @param orderId If orderId is set, it will get orders &gt;= that orderId. Otherwise, most recent orders are returned.
 *                    If startTime and/or endTime provided, orderId is not required.
 */
data class AllOrdersParams(
symbol : String, Long orderId):Params
{

	/**
	 * Creates an instance of {@link AllOrdersParams}.
	 * 
	 * @param symbol  Trading pair we want the orders.
	 * @param orderId If orderId is set, it will get orders &gt;= that orderId.
	 *                Otherwise, most recent orders are returned.
	 *                If startTime and/or endTime provided, orderId is not required.
	 */
	public AllOrdersParams(String symbol, Long orderId) {
		this.symbol = symbol;
		this.orderId = orderId;
	}

	/**
	 * Creates an instance of {@link AllOrdersParams}.
	 * 
	 * @param symbol Trading pair we want the orders.
	 */
	public AllOrdersParams(String symbol) {
		this(symbol, null);
	}
}

File: src/main/kotlin/com/binance4j/connectors/spot/param/AllOrdersParams.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.connectors.spot.param

import com.binance4j.connectors.core.param.Params
import com.binance4j.connectors.spot.client.SpotClient

/**
 * [SpotClient.getAllOrders] params.
 *
 * @param symbol  Trading pair we want the orders.
 * @param orderId If orderId is set, it will get orders &gt;= that orderId. Otherwise, most recent orders are returned.
 * If startTime and/or endTime provided, orderId is not required.
 * @param startTime Start time in ms.
 * @param endTime   End time in ms.
 * @param limit     Results limit.
 */
data class AllOrdersParams @JvmOverloads constructor(
	var symbol: String,
	var orderId: Long? = null,
	var startTime: Long? = null,
	var endTime: Long? = null,
	var limit: Int? = null
) : Params